vars/notifyBuildResult.groovy

```diff
@@ -77,56 +77,26 @@ def call(Map args = [:]) {
         data['disableGHComment'] = aggregateComments
         def notifications = []
 
-        def notificationManager = new NotificationManager()
-        if(shouldNotify && !to?.empty){
-          log(level: 'DEBUG', text: 'notifyBuildResult: Notifying results by email.')
-          notificationManager.notifyEmail(data)
-        }
+        notifyEmail(data: data, when: (shouldNotify && !to?.empty))
 
-        newPRComment.findAll { k, v ->
-          catchError(message: "There were some failures when generating the customise comment for $k", buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
-            unstash v
-            notificationManager.customPRComment(commentFile: k, file: v)
-          }
-        }
+        addGitHubCustomComment(newPRComment: newPRComment)
 
         // Should notify if it is a PR and it's enabled
-        if(notifyPRComment && isPR()) {
-          log(level: 'DEBUG', text: "notifyBuildResult: Notifying results in the PR.")
-          catchError(message: "There were some failures when notifying results in the PR", buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
-            def prComment = notificationManager.notifyPR(data)
-            notifications << prComment
-          }
+        def prComment = createGitHubComment(when: (notifyPRComment && isPR()), data: data)
+        if (prComment) {
+          notifications << prComment
         }
 
         // Should analyze flakey but exclude it when aborted
-        if(analyzeFlakey && currentBuild.currentResult != 'ABORTED') {
-          data['es'] = es
-          data['es_secret'] = secret
-          data['flakyReportIdx'] = flakyReportIdx
-          data['flakyThreshold'] = flakyThreshold
-          log(level: 'DEBUG', text: "notifyBuildResult: Generating flakey test analysis.")
-          catchError(message: "There were some failures when generating flakey test results", buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
-            def flakyComment = notificationManager.analyzeFlakey(data)
-            notifications << flakyComment
-          }
+        def flakyComment = analyzeFlaky(when: (analyzeFlakey && currentBuild.currentResult != 'ABORTED'),
+                                        data: data, es: es, secret: secret, flakyReportIdx: flakyReportIdx, flakyThreshold: flakyThreshold)
+        if (flakyComment) {
+          notifications << flakyComment
```

Comment:
I'm not sure if this works as expected, since this refactoring was done as a prerequisite of https://github.com/elastic/apm-pipeline-library/pull/916. I won't merge this PR until I'm confident enough with my manual testing